packages/rocketchat-api/server/v1/rooms.js

```diff
@@ -115,3 +115,23 @@ RocketChat.API.v1.addRoute('rooms.saveNotification', { authRequired: true }, {
 		return RocketChat.API.v1.success();
 	}
 });
+
+RocketChat.API.v1.addRoute('rooms.favorite/:roomId', { authRequired: true }, {
+	post() {
+		const { favorite } = this.bodyParams;
+		const { roomId } = this.urlParams;
```

Comment:
Why is "roomId" in the url path and "favorite" in the body? Why not everything in the body? 🤔

Like:
https://rocket.chat/docs/developer-guides/rest-api/channels/invite/

Reply:
I tried to follow what was instructed [here](https://forums.rocket.chat/t/rest-api-for-favoriting-room/465/5), but we can put it in the body, no problem

Reply:
If you'll notice the suggestion you linked to, both the roomId and the favoriting was in the query parameters and not the body or the url path.